xdsl/dialects/experimental/stencil.py

```diff
@@ -200,13 +111,90 @@ def max(a: IndexAttr, b: IndexAttr | None) -> IndexAttr:
     def as_tuple(self) -> tuple[int, ...]:
         return tuple(e.data for e in self.array.data)
 
+    def as_list(self) -> list[int]:
+        return list(e.data for e in self.array.data)
+
     def __len__(self):
         return len(self.array)
 
     def __iter__(self) -> Iterator[int]:
         return (e.data for e in self.array.data)
 
 
+class StencilType(Generic[_FieldTypeElement], ParametrizedAttribute, TypeAttribute):
+    shape: ParameterDef[IndexAttr]
+    element_type: ParameterDef[_FieldTypeElement]
+
+    def get_num_dims(self) -> int:
+        return len(self.shape)
+
+    def get_shape(self) -> list[int]:
```

Comment:
```suggestion
    def get_shape(self) -> tuple[int]:
```
If we're not planning to mutate the value, passing around tuples seems like a better call a priori, since they're more efficient and have the non-mutability guarantee

Reply:
This is a soft contract we have with memref currently. Would you prefer that I switch the memref one too as part of this PR? (This means switching uses and cluttering this PR a bit, but I don't mind myself)

Reply:
🙏